eox_hooks/actions.py

```diff
@@ -154,3 +163,71 @@ def trigger_enrollments_creation(**kwargs):
         user.username,
         followup_enrollments
     )
+
+
+def trigger_grades_assignment(**kwargs):
+    """
+    Custom action that propagates grades to a course component.
+
+    After a user's certificate generation, this action ensures that the grade
+    obtained in a course belonging to a Course Program is mapped into the
+    component that represents the course in the program.
+
+    If the course does not have the setting EDNX_TRIGGER_GRADES_ASSIGNMENT
+    in its other_course_settings, then the grade is not propagated.
+
+    The setting should look like this:
+
+    {
+        "EDNX_TRIGGER_GRADES_ASSIGNMENT": {
+            "block_id": "467f8ab131634e52bb6c22b60940d857",
+            "program_id": "course-v1:Demo+CSTest+2020",
+            "exact_score": true
+        }
+    }
+
+    For example:
+
+    Program A has three courses X, Y and Z. Then, this course has three xblocks
+    -of type staffgradedxblock- that represent each course: x, y and z respectively.
+
+    If exact score is true:
+        User `U` obtains 90% of the grade in X, then 0.9 is assigned to the component x.
+    Otherwise:
+        User `U` obtains 90% of the grade in X, then 1 is assigned to the component x.
+
+    Keyword args:
+        certificate (GeneratedCertificate): certificate generated by the course.
+    """
+    certificate = kwargs.get("certificate")
+    course = _get_course(certificate.course_id)
+
+    if not hasattr(course, "other_course_settings"):
+        return
+
+    grades_assignment_by_block = course.other_course_settings.get("EDNX_TRIGGER_GRADES_ASSIGNMENT")
+
+    program_id = grades_assignment_by_block.get("program_id")
```

Comment:
in case "EDNX_TRIGGER_GRADES_ASSIGNMENT" is not defined in other_course_settings, and grades_assignment_by_block is None, wouldn't this raise an exception?

Reply:
I thought I tested it in this line https://github.com/eduNEXT/eox-hooks/pull/23/files#diff-9a834b22a7d15565aa0e319f90e064f0aaee76cad185384cf798f098cad6b838R225 
But I didn't do it quite right. Thanks

Reply:
I didn't get _same as block_id_, in line 212 returns if block_id is None

Reply:
@mariajgrimaldi I meant that in case of grades_assignment_by_block being None, line 211 would also raise an exception